service/policy/db/registered_resources.go

```diff
@@ -277,6 +281,56 @@ func (c PolicyDBClient) GetRegisteredResourceValue(ctx context.Context, r *regis
 	}, nil
 }
 
+func (c PolicyDBClient) GetRegisteredResourceValuesByFQN(ctx context.Context, r *registeredresources.GetRegisteredResourceValuesByFQNRequest) (map[string]*policy.RegisteredResourceValue, error) {
+	resp := make(map[string]*policy.RegisteredResourceValue)
+	count := 0
+
+	for _, fqn := range r.GetFqns() {
+		normalizedFQN := strings.ToLower(fqn)
+		resp[normalizedFQN] = nil
+
+		parsed, err := util.ParseRegisteredResourceValueFqn(fqn)
+		if err != nil {
+			return nil, err
+		}
+
+		rv, err := c.Queries.getRegisteredResourceValue(ctx, getRegisteredResourceValueParams{
+			Name:  parsed.Name,
+			Value: parsed.Value,
+		})
+		if err != nil {
+			if errors.Is(err, pgx.ErrNoRows) {
```

Comment:
Is there a better way to do this? Move the pgx import to the `service/pkg/db/errors.go` file so that this DB client code isn't dependent on the provider implementation? Or change the SQL query to use `LIMIT 1` or something (if possible)?

Reply:
I'd suggest trying to refactor the SQL query. I think you could do it with a single roundtrip to surface the error?

Something like this: 
```sql
WITH name_value_pairs AS (
    SELECT unnest(:names::text[]) AS name, 
           unnest(:values::text[]) AS value
)
SELECT 
    rr.id AS resource_id, 
    rr.name AS resource_name,
    rrv.id AS value_id,
    rrv.value
FROM registered_resources rr
JOIN registered_resource_values rrv ON rr.id = rrv.registered_resource_id
JOIN name_value_pairs nvp ON rr.name = nvp.name AND rrv.value = nvp.value;
```